avocado/utils/network/interfaces.py

```diff
@@ -51,19 +51,28 @@ def __init__(self, if_name, host, if_type="Ethernet"):
     def config_filename(self):
         current_distro = distro_detect()
         if current_distro.name in ["rhel", "fedora"]:
-            path = "/etc/sysconfig/network-scripts"
+            if current_distro.name == 'rhel' and current_distro.version == '9':
+                path = "/etc/NetworkManager/system-connections"
+            else:
+                path = "/etc/sysconfig/network-scripts"
         elif current_distro.name == "SuSE":
             path = "/etc/sysconfig/network"
         else:
             msg = "Distro not supported by API. Could not get interface filename."
             raise NWException(msg)
-        return f"{path}/ifcfg-{self.name}"
+        if current_distro.name == 'rhel' and current_distro.version == '9':
+            return f"{path}/{self.name}.nmconnection"
+        else:
+            return f"{path}/ifcfg-{self.name}"
 
     @property
     def config_file_path(self):
         current_distro = distro_detect()
         if current_distro.name in ["rhel", "fedora"]:
-            return "/etc/sysconfig/network-scripts"
+            if current_distro.name == 'rhel' and current_distro.version == '9':
+                return "/etc/NetworkManager/system-connections"
+            else:
+                return "/etc/sysconfig/network-scripts"
```

Comment:
if current_distro.name == 'rhel' and current_distro.version == '9':    this is being checked at multiple places in same file .. can this be optimised ?